Simplenote/src/main/java/com/automattic/simplenote/NotesActivity.java

```diff
@@ -1061,45 +1058,47 @@ private void updateActionsForLargeLandscape(Menu menu) {
         menu.findItem(R.id.menu_empty_trash).setVisible(mSelectedTag != null && mSelectedTag.id == TRASH_ID);
     }
 
-    public void updateViewsAfterTrashAction(Note note) {
+    public void trashNote(Note note) {
         if (note == null || isFinishing()) {
             return;
         }
 
+        note.setDeleted(!note.isDeleted());
+        note.setModificationDate(Calendar.getInstance());
+        note.save();
+
         if (note.isDeleted()) {
             List<String> deletedNoteIds = new ArrayList<>();
             deletedNoteIds.add(note.getSimperiumKey());
             mUndoBarController.setDeletedNoteIds(deletedNoteIds);
             mUndoBarController.showUndoBar(getUndoView(), getString(R.string.note_deleted));
             AnalyticsTracker.track(
-                LIST_NOTE_DELETED,
-                CATEGORY_NOTE,
-                "overflow_menu"
+                    LIST_NOTE_DELETED,
+                    CATEGORY_NOTE,
+                    "overflow_menu"
             );
         } else {
             AnalyticsTracker.track(
-                EDITOR_NOTE_RESTORED,
-                CATEGORY_NOTE,
-                "overflow_menu"
+                    EDITOR_NOTE_RESTORED,
+                    CATEGORY_NOTE,
+                    "overflow_menu"
             );
         }
 
-        // If we just deleted/restored the active note, show the placeholder
-        if (mCurrentNote != null && mCurrentNote.getSimperiumKey().equals(note.getSimperiumKey())) {
-            showDetailPlaceholder();
-        }
-
-        NoteListFragment fragment = getNoteListFragment();
-        if (fragment != null) {
+        if(getNoteListFragment() != null) {
+            NoteListFragment fragment = getNoteListFragment();
+            if (DisplayUtils.isLargeScreenLandscape(this)) {
+                fragment.updateSelectionAfterTrashAction();
+            }
             fragment.getPrefs();
             fragment.refreshList();
         }
 
         if (mInvalidateOptionsMenuHandler != null) {
             mInvalidateOptionsMenuHandler.removeCallbacks(mInvalidateOptionsMenuRunnable);
             mInvalidateOptionsMenuHandler.postDelayed(
-                mInvalidateOptionsMenuRunnable,
-                getResources().getInteger(android.R.integer.config_shortAnimTime)
+                    mInvalidateOptionsMenuRunnable,
+                    getResources().getInteger(android.R.integer.config_shortAnimTime)
```

Comment:
```suggestion
                mInvalidateOptionsMenuRunnable,
                getResources().getInteger(android.R.integer.config_shortAnimTime)
```

This might have been automatically edited by Android Studio.  The edit and extra indention is not necessary.